contrib/database/sql/propagation_test.go

```diff
@@ -20,121 +24,164 @@ import (
 	"github.com/stretchr/testify/require"
 )
 
+type dummyTransport struct{}
+
+func (d *dummyTransport) RoundTrip(_ *http.Request) (*http.Response, error) {
+	return &http.Response{
+		Body: io.NopCloser(bytes.NewBufferString("{}")),
+	}, nil
+}
+
+type noopLogger struct{}
+
+func (n *noopLogger) Log(_ string) {
+	return
+}
+
 func TestDBMPropagation(t *testing.T) {
 	// Ensure the global service name is set to the previous value after we finish the test, since the
 	// tracer.WithService option overrides it.
 	prevServiceName := globalconfig.ServiceName()
 	defer globalconfig.SetServiceName(prevServiceName)
 
 	testCases := []struct {
-		name     string
-		opts     []RegisterOption
-		callDB   func(ctx context.Context, db *sql.DB) error
-		prepared []string
-		executed []*regexp.Regexp
+		name           string
+		tracerOpts     []tracer.StartOption
+		registerOpts   []RegisterOption
+		opts           []Option
+		skipParentSpan bool
+		callDB         func(ctx context.Context, db *sql.DB) error
+		prepared       []string
+		executed       []*regexp.Regexp
```

Comment:
Can you please elaborate what are the changes in testing about? I see that they don't touch the code changed in ddtrace/tracer files